alibi/explainers/tests/test_cfproto.py

```diff
@@ -1,89 +1,20 @@
 # flake8: noqa E731
 import numpy as np
 import pytest
-from sklearn.datasets import load_iris
-from sklearn.preprocessing import OneHotEncoder
 import tensorflow as tf
-from tensorflow.keras.utils import to_categorical
 import keras
 from alibi.api.defaults import DEFAULT_META_CFP, DEFAULT_DATA_CFP
 from alibi.datasets import fetch_adult
 from alibi.explainers import CounterFactualProto
 from alibi.utils.mapping import ord_to_ohe, ohe_to_ord, ord_to_num
 
-
-@pytest.fixture
-def tf_keras_iris_model(request):
-    if request.param == 'keras':
-        k = keras
-    elif request.param == 'tf':
-        k = tf.keras
-    else:
-        raise ValueError('Unknown parameter')
-
-    x_in = k.layers.Input(shape=(4,))
-    x = k.layers.Dense(10, activation='relu')(x_in)
-    x_out = k.layers.Dense(3, activation='softmax')(x)
-    model = k.models.Model(inputs=x_in, outputs=x_out)
-    model.compile(loss='sparse_categorical_crossentropy', optimizer='sgd', metrics=['accuracy'])
-    return model
-
-
-@pytest.fixture
-def tf_keras_iris_ae(request):
-    if request.param == 'keras':
-        k = keras
-    elif request.param == 'tf':
-        k = tf.keras
-    else:
-        raise ValueError('Unknown parameter')
-
-    # encoder
-    x_in = k.layers.Input(shape=(4,))
-    x = k.layers.Dense(5, activation='relu')(x_in)
-    encoded = k.layers.Dense(2, activation=None)(x)
-    encoder = k.models.Model(x_in, encoded)
-
-    # decoder
-    dec_in = k.layers.Input(shape=(2,))
-    x = k.layers.Dense(5, activation='relu')(dec_in)
-    decoded = k.layers.Dense(4, activation=None)(x)
-    decoder = k.models.Model(dec_in, decoded)
-
-    # autoencoder = encoder + decoder
-    x_out = decoder(encoder(x_in))
-    autoencoder = k.models.Model(x_in, x_out)
-    autoencoder.compile(optimizer='adam', loss='mse')
-
-    return autoencoder, encoder, decoder
-
-
-@pytest.fixture
-def tf_keras_iris(tf_keras_iris_model, tf_keras_iris_ae):
-    X, y = load_iris(return_X_y=True)
-    X = (X - X.mean(axis=0)) / X.std(axis=0)  # scale dataset
-
-    idx = 145
-    X_train, y_train = X[:idx, :], y[:idx]
-    # y_train = to_categorical(y_train) # TODO: fine to leave as is?
-
-    # set random seed
-    np.random.seed(1)
-    tf.set_random_seed(1)
-
-    model = tf_keras_iris_model
-    model.fit(X_train, y_train, batch_size=128, epochs=500, verbose=0)
-
-    ae, enc, _ = tf_keras_iris_ae
-    ae.fit(X_train, X_train, batch_size=32, epochs=100, verbose=0)
-
-    return X_train, model, ae, enc
+from alibi_test_models.data import adult_data, iris_data
 
 
 @pytest.fixture
-def tf_keras_iris_explainer(request, tf_keras_iris):
-    X_train, model, ae, enc = tf_keras_iris
-
+def tf_keras_iris_explainer(request, models, get_iris_dataset):
```

Comment:
I'm not sure I follow where `get_iris_dataset` is used here? It seems that `iris_data()` returns the data instead?

Reply:
Well spotted